lib/runners/processor/detekt.rb

```diff
@@ -0,0 +1,436 @@
+
+    def switch_constructor(report_id, output)
+      case report_id
+      when "html"
+        construct_html_result(output)
+      when "xml"
+        construct_checkstyle_result(output)
+      when "txt"
+        construct_plain_result(output)
+      end
+    end
+
+    def construct_html_result(output)
+      issues = []
+
+      # NOTE: HTML that Detekt returns unable to parse..
+      regexp_id = /<details id="(.*)" open=.*/
+      regexp_detail = /<li><span class="location">(.*):(.*):(.*)<\/span><br><span class="message">(.*)<\/span>/
+      
+      rule = nil
+      output.lines(chomp: true).map do |line|
+        
+        lrule = line.match(regexp_id)
+        unless lrule.nil?
+          rule, = lrule.captures
+          next
+        end
+
+        ldetail = line.match(regexp_detail)
+        unless ldetail.nil?
+          file, start_line, _col, message = ldetail.captures
+          issues << construct_issue(
+            file: file,
+            line: start_line,
+            message: message,
+            rule: rule,
+          )
+        end
+      end
+
+      issues
+    end
+
+    def construct_checkstyle_result(output)
+      document = REXML::Document.new(output)
+      unless document.root
+        msg = "Invalid XML output!"
+        trace_writer.error msg
+        raise msg
+      end
+
+      issues = []
+
+      document.root.each_element("file") do |file|
+        file.each_element do |error|
+          case error.name
+          when "error"
+            issues << construct_issue(
+              file: file[:name],
+              line: error[:line],
+              message: error[:message],
+              rule: error[:source],
+            )
+          else
+            add_warning error.text.strip, file: file[:name]
+          end
+        end
+      end
+
+      issues
+    end
+
+    def construct_plain_result(output)
+      issues = []
+
+      regexp = /(.*) at (.*):(.*):(.*) - (.*)/
+
+      output.lines(chomp: true).map do |line|
+        match = line.match(regexp)
+
+        unless match.nil?
+          rule, file, start_line, _, message = match.captures
+          issues << construct_issue(
+            file: file,
+            line: start_line,
+            message: message,
+            rule: rule,
+          )
+        end
+      end
+
+      issues
+    end
+
+    def construct_issue(file:, line:, message:, rule:)
+      Issue.new(
+        path: relative_path(working_dir.realpath / file, from: working_dir.realpath),
+        location: Location.new(start_line: line),
+        id: rule,
+        message: message,
+      )
+    end
+
+    def cli_baseline
+      unless cli_config[:baseline].nil?
+        return check_user_file_exists(
+          [cli_config[:baseline]],
+          "baseline"
+        )
+      end
+    end
+
+    def cli_config_files
+      unless cli_config[:config].empty?
+        return check_user_file_exists(
+          cli_config[:config],
+          "config"
+        )
+      end
```

Comment:
[suggestion] Same here: https://github.com/sider/runners/pull/749/files#r384348305

```suggestion
      cli_config[:config].then { |value| value.present? ? ["--config", value.join(",")] : [] }
```